rest-api/dao/participant_summary_dao.py

```diff
@@ -379,6 +381,77 @@ def to_client_json(self, model):
 
     return result
 
+  def make_csv(self, results):
+    csv_data = StringIO.StringIO()
+    writer = unicode_csv.UnicodeWriter(csv_data)
+    headers_list = []
+    write_headers = True
+    for row in results.items:
+      line = self.to_client_csv(row)
+      values_list = []
+
+      if len(headers_list) == 0:
+        for k in line.keys():
+          headers_list.append(k)
+      else:
+        for k in line.keys():
+          if k not in headers_list:
+            headers_list.append(k)
+            write_headers = True
+
+      for i in headers_list:
+        values_list.append(line.get(i))
+
+      if write_headers:
```

Comment:
Why not just do this check outside the loop and write the row before you get in here?

We'd want to return the headers even if we had no results... (we should have a test for that.)